File: bench/Chain.java
public class Chain {
    public static void main(String[] args) {
        long n = Long.parseLong(args[0]);
        long sum = 0;
        for (long i = 1; i <= n; i++) sum = (sum + i) % 1000000007L;
        System.out.println(sum);
    }
}

File: bench/chain.rs
fn main() {
    let n: i64 = std::env::args().nth(1).unwrap().parse().unwrap();
    let mut sum: i64 = 0;
    let mut i: i64 = 1;
    while i <= n { sum = (sum + i) % 1_000_000_007; i += 1; }
    println!("{}", std::hint::black_box(sum));
}
